===== Patch instructions generated by SAVER =====
[+] { Insert: if ({ p11_filter_deny_token:n$9 == { 0 } }) free(*(p11_filter_deny_token:token_copy)) at 9 (line 431, column 7) }
===== Actual patch in diff format =====
@@ -429,6 +429,7 @@ p11_filter_deny_token (p11_virtual *virt,
 	return_if_fail (token_copy != NULL);
 
 	if (!p11_array_push (filter->entries, token_copy))
+		free(token_copy);
 		return_if_reached ();
 
 	if (filter->initialized)
